api/envoy/extensions/filters/network/filter_state_dynamic_forward_proxy/v3/filter_state_dynamic_forward_proxy.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.network.filter_state_dynamic_forward_proxy.v3;
+
+import "envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.network.filter_state_dynamic_forward_proxy.v3";
+option java_outer_classname = "SniDynamicForwardProxyProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/filter_state_dynamic_forward_proxy/v3;filter_state_dynamic_forward_proxyv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Filter State dynamic forward proxy]
+
+message FilterConfig {
+  // The DNS cache configuration that the filter will attach to. Note this
+  // configuration must match that of associated :ref:`dynamic forward proxy
+  // cluster configuration
+  // <envoy_v3_api_field_extensions.clusters.dynamic_forward_proxy.v3.ClusterConfig.dns_cache_config>`.
+  common.dynamic_forward_proxy.v3.DnsCacheConfig dns_cache_config = 1
+      [(validate.rules).message = {required: true}];
+
+  oneof port_specifier {
+    // The port number to connect to the upstream.
+    uint32 port_value = 2 [(validate.rules).uint32 = {lte: 65535 gt: 0}];
```

Comment:
we have `port_value` because SNI doesn't contain port. You might want an option to retrieve the port from filter state as well.